### PR TITLE
Feature/add routes to tabs

### PR DESCRIPTION
**Pull request by tomixtomi12**
+ add route for each tab
+ add custom route transitions since md-tab-link and md-tab-nav don't animate the content

### CODE REVIEW

File: businesslogic/src/main/java/eu/execom/teampomodoro/businesslogic/implementation/ReportServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        result.setEmail(user.getEmail());
+        result.setAveragePomodoroDuration(calculateAveragePomodoroDuration(pomodoros));
+        result.setContributionPercentage(calculateContributionPercentage(pomodoros));
+        result.setNumberOfPomodoros(pomodoros.size());
+
+        return result;
+    }
+
+    private Double calculateContributionPercentage(List<Pomodoro> usersPomodoros) {
+        if (usersPomodoros.size() == 0) return 0D;
```
Comment (asuletic) [inline businesslogic/src/main/java/eu/execom/teampomodoro/businesslogic/implementation/ReportServiceImpl.java]: Use brackets for if body statements.

File: web/src/main/webapp/src/app/shared/time.pipe.ts [hunk 2]
```diff
@@ -6,13 +6,24 @@ import {Pipe, PipeTransform} from '@angular/core';
 export class TimePipe implements PipeTransform {
 
   transform(value: number): string {
+    if (value === 0) return '00:00:00';
```
Comment (asuletic) [inline web/src/main/webapp/src/app/shared/time.pipe.ts]: Use brackets for body in if statement.

File: web/src/main/webapp/src/app/task/task/task.component.html [hunk 3]
```diff
@@ -20,22 +20,24 @@ <h1 class="title">Tasks</h1>
...
     </tr>
     <tr *ngFor="let task of tasks; let i = index">
-      <td>{{task.id}}</td>
-      <td>{{task.description}}</td>
+      <td class="middle">{{task.id}}</td>
+      <td class="middle">{{task.description}}</td>
+      <td class="middle">{{task.pomodoroCountEstimate}}</td>
       <td>
-        <button md-button class="btn btn-warning" (click)="editTask(i)">Edit task</button>
+        <button md-button [disabled]="task.description === 'break'" class="btn btn-warning" (click)="editTask(i)">Edit task</button>
```
Comment (asuletic) [inline web/src/main/webapp/src/app/task/task/task.component.html]: NOTE: Check for 'disabled' will be changed when you remove break task. This issue is linked with one mentioned above in ReportServiceImpl.

File: web/src/main/webapp/src/app/pomodoro/pomodoro-history/pomodoro-history.component.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<div class="container">
+  <h1 class="title">Pomodoro history</h1>
+
+  <br><br>
```
Comment (tomixtomi12) [inline web/src/main/webapp/src/app/pomodoro/pomodoro-history/pomodoro-history.component.html]: Currently, all components are using line breaks, I will fix this on a cleanup branch.

**Pull request merged**